cmd/earthly/flag/global.go

```diff
@@ -517,12 +516,5 @@ func (global *Global) RootFlags(installName string, bkImage string) []cli.Flag {
 			Usage:       "A remote docker image tag use as explicit cache",
 			Destination: &global.RemoteCache,
 		},
-		&cli.BoolFlag{
-			Name:        "use-remote-registry",
-			EnvVars:     []string{"EARTHLY_USE_REMOTE_REGISTRY"},
-			Usage:       "Use remote registry with Docker pull",
-			Destination: &global.UseRemoteRegistry,
-			Hidden:      true, // Internal.
-		},
```

Comment:
I wonder if we should keep this flag around (and just flipping the default value), just in case a user needs to turn it off for some reason (e.g. running into a bug)?

Reply:
I'm on the fence regarding this, yes it's a nice escape hatch; however keeping around all these old features adds to maintenance cost.

We could leave it as-is for a few releases, and then fully remove it down the road; alternatively if a user runs into the need for this, we could re-introduce it, and the user could pin to an older version until it's done.

Reply:
Yeah, that's reasonable. Let's nuke it then

Reply:
@alexcb and I were talking about this as well. On the one hand, it could be useful to retain the option to disable it. On the other, there's a small, accumulating cost with keeping these around (code, tests, etc.).

I'd vote for removing it and fixing bugs as they come up. All the tests are green :green_apple: